### PR TITLE
Add test app

### PR DESCRIPTION
**Pull request by 1bsyl**
Add test app / fuzzer for SDL_ttf, 
this the test app from https://github.com/libsdl-org/SDL_ttf/issues/98
that I use for find crash / bug with SDL_ttf

### CODE REVIEW

File: testapp.c [hunk 1]
```diff
@@ -0,0 +1,1232 @@
...
+/* RENDER_SOLID = 0, RENDER_BLENDED = 1, RENDER_SHADED = 2, RENDER_LCD = 3 } */
+static int render_mode = -1;
+static int render_mode_overwrite;
+static int render_mode_count = 4;
+static const char *render_mode_desc[] = { "Solid", "Blended", "Shaded", "LCD" };
+
+static int direction = 0;
+static int direction_count = 4;  
+static const char *direction_desc[] = { "LTR", "RTL", "TTB", "BTT"  };
+static const int  *direction_val[] = { TTF_DIRECTION_LTR, TTF_DIRECTION_RTL, TTF_DIRECTION_TTB, TTF_DIRECTION_BTT };
```
Comment (madebr) [inline testapp.c:164]:
```suggestion
static const int   direction_val[] = { TTF_DIRECTION_LTR, TTF_DIRECTION_RTL, TTF_DIRECTION_TTB, TTF_DIRECTION_BTT };
```

Comment (1bsyl) [reply testapp.c]: good catch ! done

**Comment by madebr** [comment]
Could you upload a new fonts archive somewhere?
Because the one from https://github.com/libsdl-org/SDL_ttf/issues/98#issue-806097965 does not contain `NotoColorEmoji.ttf` (and possibly more).

**Comment by 1bsyl** [comment]
not sure if we need to provide fonts.
this can be used with any font. 
there was an idea to scan a "./fonts" to find fonts.
but sdl doesn't provide directory listing (yet)

File: testapp.c [hunk 2]
```diff
@@ -0,0 +1,1232 @@
...
+#if 0
+# include "fonts/fuzzing_corpora.h"
+#endif
+#if 0
+# include "fonts/android_fonts_others.h"
+# include "fonts/android_fonts.h"
+#endif
+    "fonts/Kaumudi.ttf"
+};
+static const int test_fonts_count = sizeof(test_fonts)/sizeof(char*);
```
Comment (madebr) [inline testapp.c:132]: You can use `SDL_arraysize` here, and in other locations.

File: testapp.c [hunk 3]
```diff
@@ -0,0 +1,1232 @@
...
+/* RENDER_SOLID = 0, RENDER_BLENDED = 1, RENDER_SHADED = 2, RENDER_LCD = 3 } */
+static int render_mode = -1;
+static int render_mode_overwrite;
+static int render_mode_count = 4;
+static const char *render_mode_desc[] = { "Solid", "Blended", "Shaded", "LCD" };
+
+static int direction = 0;
+static int direction_count = 4;  
+static const char *direction_desc[] = { "LTR", "RTL", "TTB", "BTT"  };
+static const int  direction_val[] = { TTF_DIRECTION_LTR, TTF_DIRECTION_RTL, TTF_DIRECTION_TTB, TTF_DIRECTION_BTT };
```
Comment (madebr) [inline testapp.c:164]: I would do this as:
```c
static struct {
    const char *description;
    int value;
} directions[] = {
    { "LTR", TTF_DIRECTION_LTR },
    { "RTL", TTF_DIRECTION_RTL },
    { "TTB", TTF_DIRECTION_TTB },
    { "BTT", TTF_DIRECTION_BTT }
};
```
And use `SDL_arraysize` for `direction_count`.

Comment (1bsyl) [reply testapp.c]: yes done!
this was a test app that evolved occasionally,
there is definitively some room for improvement !

File: testapp.c [hunk 4]
```diff
@@ -0,0 +1,1240 @@
...
+    { "RTL", TTF_DIRECTION_RTL },
+    { "TTB", TTF_DIRECTION_TTB },
+    { "BTT", TTF_DIRECTION_BTT }
+};
+static int direction_count = SDL_arraysize(directions);
+
+
+//static const char *hinting_desc[] = { "normal", "light", "light_subpix", "lcd_subpix", "mono", "none" };
+static const char *hinting_desc[] = { "normal", "light", "light_subpix", "mono", "none" };
+static int hinting_count = SDL_arraysize(hinting_desc);
```
Comment (madebr) [inline testapp.c:176]:
```suggestion
static const int hinting_count = SDL_arraysize(hinting_desc);
```

File: testapp.c [hunk 5]
```diff
@@ -0,0 +1,1240 @@
...
+static int seed = 0;
+static int print_elapsed_ticks = 0;
+static int update_screen_mode = 0;
+static int save_to_bmp = 0;
+
+/* RENDER_SOLID = 0, RENDER_BLENDED = 1, RENDER_SHADED = 2, RENDER_LCD = 3 } */
+static int render_mode = -1;
+static int render_mode_overwrite;
+static const char *render_mode_desc[] = { "Solid", "Blended", "Shaded", "LCD" };
+static int render_mode_count = SDL_arraysize(render_mode_desc);
```
Comment (madebr) [inline testapp.c:159]:
```suggestion
static const int render_mode_count = SDL_arraysize(render_mode_desc);
```

Comment (1bsyl) [reply testapp.c]: updated!

**Comment by 1bsyl** [comment]
Re-based and fixed build of testapp.c

**Pull request merged**